lib/core/src/lib/form/components/form-field/form-field.component.ts

```diff
@@ -88,14 +92,29 @@ export class FormFieldComponent implements OnInit, OnDestroy {
                     instance.fieldChanged.subscribe((field) => {
                         if (field && this.field.form) {
                             this.visibilityService.refreshVisibility(field.form);
-                            field.form.onFormFieldChanged(field);
+                            this.triggerFormFieldChanged(field);
                         }
                     });
+
+                    if (FormFieldTypes.isReactiveType(instance?.field?.type)) {
+                        this.updateReactiveFormControlOnFormRulesEvent(instance);
+                    }
                 }
             }
         }
     }
 
+    private updateReactiveFormControlOnFormRulesEvent(instance: any): void {
+        instance?.formService.formRulesEvent.pipe(takeUntilDestroyed(this.destroyRef)).subscribe(() => {
```

Comment:
Reducing the check to only the form rules might miss some other parts where we update the form and the field changed event is triggered

Reply:
on each onFieldChanged call, inside the [base widget.component.ts ](https://github.com/Alfresco/alfresco-ng2-components/blob/develop/lib/core/src/lib/form/components/widgets/widget.component.ts#L106), and also on the all events, we update the value of formRulesEvent subject so it will be always triggered for all the events (and its needed for all events, because in modeling, we can set rules for all available event types, like blur, click etc). i think it should be enough, but maybe im not aware of some corner case